### PR TITLE
Add LLVM 11 as a freedesktop SDK extension

### PR DESCRIPTION
**Pull request by Lctrs**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Not needed**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *Patches took from freedesktop SDK or already applied upstream*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

Basically, I tried to stick to https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/master/elements/components/llvm.bst as much as possible.

Should be merged in `branch/20.08`.

Needed for building newer versions of `net.rpcs3.RPCS3`.

### CODE REVIEW

**Comment by Lctrs** [comment]
bot build org.freedesktop.Sdk.Extension.llvm11

**Comment by Lctrs** [comment]
bot build org.freedesktop.Sdk.Extension.llvm11

**Comment by barthalion** [comment]
Actually let's see if it builds first:

bot, build org.freedesktop.Sdk.Extension.llvm11

**Comment by Lctrs** [comment]
@barthalion It doesn't. I'm working on it.

**Comment by Lctrs** [comment]
bot, build org.freedesktop.Sdk.Extension.llvm11

**Comment by Lctrs** [comment]
Hmm @barthalion Any idea for the license ?

**Comment by barthalion** [comment]
Maybe this? https://spdx.org/licenses/LLVM-exception.html

**Comment by Lctrs** [comment]
Nevermind I found it. Will push a fix.

**Comment by Lctrs** [comment]
bot, build org.freedesktop.Sdk.Extension.llvm11

**Comment by Lctrs** [comment]
@barthalion Still no luck. According to https://spdx.github.io/spdx-spec/appendix-IV-SPDX-license-expressions/#3-exception-with-operator, `Apache-2.0 WITH LLVM-exception` is a valid SPDX ID though.

**Comment by Lctrs** [comment]
@barthalion Seems like it's not supported by https://github.com/hughsie/appstream-glib, so I've skipped appstream check for now.

bot, build org.freedesktop.Sdk.Extension.llvm11

**Comment by Lctrs** [comment]
Seems like `*.so` files are not stripped. Is it because they are in a subdirectory of `lib` ?

**Comment by Lctrs** [comment]
bot, build org.freedesktop.Sdk.Extension.llvm11

**Comment by Lctrs** [comment]
I believe no stripping is due to https://github.com/flatpak/flatpak-builder/issues/317

**Comment by Lctrs** [comment]
bot, build org.freedesktop.Sdk.Extension.llvm11

**Comment by Lctrs** [comment]
@barthalion Ready to be merge from my POV.

**Comment by Erick555** [comment]
@Lctrs @barthalion it's not ready. It's 3x times size of whole Sdk (with previous llvm included) and this is without static libs. The problem is it's being build with full debug info (`-g`) while it should use `-g1` (which is what runtime uses).

**Comment by Erick555** [comment]
Another issue is removal of all static libs while keeping headers that still refers to them which tricks some apps they can build stuff that cannot be build. The same issue happens in Sdk so it may be acceptable for the time being.

https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1147

**Comment by Lctrs** [comment]
> @Lctrs @barthalion it's not ready. It's 3x times size of whole Sdk (with previous llvm included) and this is without static libs. The problem is it's being build with full debug info (`-g`) while it should use `-g1` (which is what runtime uses).

Yeah. I don't know how to change this in flatpak. Can you tell me how ?

> Another issue is removal of all static libs while keeping headers that still refers to them which tricks some apps they can build stuff that cannot be build. The same issue happens in Sdk so it may be acceptable for the time being.
> 
> https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1147

Shall I cleanup all `/include` dir then ? That would save a trim size even more.

**Comment by Lctrs** [comment]
> > @Lctrs @barthalion it's not ready. It's 3x times size of whole Sdk (with previous llvm included) and this is without static libs. The problem is it's being build with full debug info (`-g`) while it should use `-g1` (which is what runtime uses).
> 
> Yeah. I don't know how to change this in flatpak. Can you tell me how ?
> 

Nevermind. Found it.

bot, build org.freedesktop.Sdk.Extension.llvm11

**Comment by Erick555** [comment]
> Shall I cleanup all /include dir then ? That would save a trim size even more.

No, you would need to examine all headers and remove only unusable ones which is quite burdensome. Alternatively you may keep some static libs (which would make some people happy) but this is again quite burdensome in flatpak. I would leave this issue for later (after initial merge) then.

**Comment by Lctrs** [comment]
@Erick555 much better. Thanks !
